autoload/lightline/bufferline.vim

```diff
@@ -55,6 +56,8 @@ function! s:get_buffer_name(i, buffer)
   endif
   if s:enable_devicons == 1 && exists('*WebDevIconsGetFileTypeSymbol')
     let l:name = WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t')) . ' ' . l:name
+  elseif s:enable_nerdfont == 1 && exists('*nerdfont#find')
```

Comment:
`exists` cannot be used for autoload function.

Reply:
> The exists cannot be used for autoload functions. The function does not exist until first call thus if this plugin become the first integration, it does not exist.

Thanks!

> Use try/catch way instead if you need to fallback silently. But I think users may want to know why they don't see a nice icon even they configured the variable by error

Agree. So, should I remove any existence check?

Reply:
I think it's up to project owner, @mengelbrecht right?

Reply:
Thanks for the PR @Shatur95 and thanks for reviewing it @lambdalisue.

When removing any existence check the whole bufferline disappears if the user has configured `enable_nerdfont` to `1` but `nerdfont.vim` is not installed. A missing bufferline indicates that something is wrong but does not hint at the underlying issue.
I think the plugin should handle it gracefully and should continue working without displaying an icon in this case. Therefore I would prefer the try/catch approach.

Reply:
Okay!

Reply:
I tried the following:

```vim
    try
      let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
    catch
    endtry
```

But If I remove `nerdfont.vim` I have an error and broken buffer

Reply:
The following does not show error on my environment without nerdfont.vim. Thus I've no idea about that.

```vim
try
  echomsg nerdfont#find(fnamemodify(bufname('%'), ':t'), 0)
catch /^Vim\%((\a\+)\)\=:E117:/
endtry
```

Reply:
The following snippet also works for me without error when nerdfont.vim is missing:
```viml
    try
      let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
    catch /^Vim\%((\a\+)\)\=:E117:/
    endtry
```

Reply:
Oh, that error was from Startify, not from Bufferline :D Thanks.